### ISSUE TITLE
Error in msmsWorkflow Step 4 (recalibrate) - with knowns

### ISSUE THREAD

**Issue by schymane** [MassBank/RMassBank#156]
Error in unlist(ms1peaks[, tonum]) : 
  error in evaluating the argument 'x' in selecting a method for function 'unlist': Error in `[.data.frame`(ms1peaks, , tonum) : undefined columns selected
warnings() 
[... 1 to 36 are no spectra retrieved, I expected that]
37: In .Method(..., deparse.level = deparse.level) :
  number of columns of result is not a multiple of vector length (arg 1)

It could be that we don't have enough data to recalibrate (it didn't find many spectra at all) but something more elegant than an error would be good ;)
Workspace & settings following by email @ermueller @meowcat

**Comment by ermueller** [comment]
I isolated the problem:
When recalibrating we need the m/z of the MS1-Peak of the compound that gets fragmented and we find this by doing:

````
# Get mz with ppm as specified in settings
mzL <- findMz.formula(cpd@formula,mode,recalibrateMS1Window,0)
# Retrieve all m/z from MS1 Spectrum
ms1 <- mz(cpd@parent)
# Which m/z are within the ppm?
mzFound <- ms1[(ms1 >= mzL$mzMin) & (ms1 <= mzL$mzMax)]
````
The problem here lies therein that there are two peaks within the specified ppm inside the MS1 Spectrum. Valid peaks would go from 
`156.9941` to `156.9988`
and we have 2 peaks inside the MS1 Spectrum that fit that description: 
`156.9955` and `156.9983`

and both of them being loaded into `mzFound` destroys the data structures further along the way, since it needs to only contain one value for everything to work.

So the question is now:
Do we take the MS1-Peak that has the lower dppm  or do we take the one with higher intensity, which one of these should be included in MS1 recalibration? It probably wouldn't make a noticable difference, but we should do it correctly anyways.

**Comment by uchem-massbank** [comment]
I’d say lowest ppm – no guarantee that the most intense one is the one we want, whereas we can hope that the measured mass is closest to the one the users put in their compound list / the one calculated from the SMILES. The latter (mass in the compound list) can always be adjusted by the user if the wrong one was chosen, whereas the most intense can’t be …

**Comment by ermueller** [comment]
The fix for this is in 95bc68b61ae4dd86c15975aa5baf083207cb7da8 :)
